docs/pages/access-controls/login-rules/reference.mdx

```diff
@@ -0,0 +1,571 @@
+uppercase.
+
+#### Arguments
+
+Argument | Type | Description
+-------- | ---- | -----------
+`input` | `set` | set of input strings to convert to uppercase |
+
+#### Returns
+
+Type | Description
+---- | ----------
+`set`  | a copy of `input` where each string has been convert to uppercase
+
+#### Examples
+
+Expression | Result
+------- | ------
+`strings.upper(set("Alice"))` | `("ALICE")`
+`strings.upper(set("AbCdE", "fGhIj))` | `("ABCDE", "FGHIJ")`
+
+### `strings.lower`
+
+#### Signature
+
+```go
+func strings.lower(input set) set
+```
+
+#### Description
+
+`strings.lower` returns a copy of the given set of strings converted to
+lowercase.
+
+#### Arguments
+
+Argument | Type | Description
+-------- | ---- | -----------
+`input` | `set` | set of input strings to convert to lowercase |
+
+#### Returns
+
+Type | Description
+---- | ----------
+`set`  | a copy of `input` where each string has been convert to lowercase
+
+#### Examples
+
+Expression | Result
+------- | ------
+`strings.lower(set("Alice"))` | `("alice")`
+`strings.lower(set("AbCdE", "fGhIj))` | `("abcde", "fghij")`
+
+### `strings.replaceall`
+
+#### Signature
+
+```go
+func strings.replaceall(input set, match string, replacement string) set
+```
+
+#### Description
+
+`strings.replaceall` implements substring replacement on sets of strings.
+The return value is a copy of `input` where each substring match of `match`
+found in each element of `input` will be replaced with `replacement`.
+The matching is literal and does not support regular expressions.
+
+#### Arguments
+
+Argument | Type | Description
+-------- | ---- | -----------
+`input` | `set` | set of input strings where replacements are needed |
+`match` | `string` | literal substring to be replaced |
+`replacement` | `string` | literal string to replace all instances of `match` |
+
+#### Returns
+
+Type | Description
+---- | ----------
+`set`  | a copy of `input` where each instance of `match` found in each element of `input` has been replaced with `replacement`.
+
+#### Examples
+
+Expression | Result
+------- | ------
+`strings.replaceall(set("user-name"), "-", "_")` | `("user_name")`
+`strings.replaceall(set("user-alice", "user-bob"), "user-", "")` | `("alice", "bob")`
+
+### `ifelse`
+
+#### Signature
+
+```go
+func ifelse(cond bool, valueIfTrue any, valueIfFalse any) any
+```
+
+#### Description
+
+`ifelse` implements the classic if-else branch in a pure functional style.
+If the first argument evaluates to `true` the second argument is returned, else
+the third argument is returned.
+
+#### Arguments
+
+Argument | Type | Description
+-------- | ---- | -----------
+`cond` | `bool` | condition that determines which of the following two arguments is returned |
+`valueIfTrue` | `any` | value to return if `cond` is true |
+`valueIfFalse` | `any` | value to return if `cond` is false |
+
+#### Returns
+
+Type | Description
+---- | ----------
+`any`  | the second or third argument will be returned identically
```

Comment:
I'm not sure what "identically" means here. Is there another way to phrase this?

Reply:
I was going for something like google's second definition for identically, but it's probably better to just remove the word
> LOGIC•MATHEMATICS
in the manner of an identity; for all values.